Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3363,8 +3366,12 @@ void MediaPlayerPrivateGStreamer::updateVideoOrientation(const GstTagList* tagLi
     if (m_videoSourceOrientation.usesWidthAsHeight())
         m_videoSize = m_videoSize.transposedSize();
 
-    callOnMainThreadAndWait([this] {
+    m_abortableTaskQueue.enqueueTaskAndWait<AbortableTaskQueue::Void>([weakThis = WeakPtr { *this }, this] {
+        if (!weakThis)
+            return AbortableTaskQueue::Void();
+
```

Comment:
There is no need for WeakPtr:

1. AbortableTaskQueue runs the tasks in the main thread.
2. All existing and future tasks in an abortable task queue are cancelled when .startAborting() is called (e.g. in the destructor). This is the whole point of AbortableTaskQueue.

By double checking here, a reader may wrongly infer AbortableTaskQueue doesn't have these characteristics.

Reply:
Thanks for the review, I'll clean this up in a follow-up patch.